packages/Webkul/Product/src/Type/AbstractType.php

```diff
@@ -659,16 +659,20 @@ public function getCustomerGroupPrice($product, $qty)
                 continue;
             }
 
-            if ($price->value < $lastPrice) {
-                if ($price->value_type == 'discount') {
-                    $lastPrice = $product->price - ($product->price * $price->value) / 100;
-                } else {
-                    $lastPrice = $price->value;
-                }
+            if ($price->value_type == 'discount') {
+                $lastPrice = $product->price - ($product->price * $price->value) / 100;
```

Comment:
Hi, this will fail because of the main case you have added for the `flat` case. Now just try to add a discount price which is greater than the product price you will get negative values.